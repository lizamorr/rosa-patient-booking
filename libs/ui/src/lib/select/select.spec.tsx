import { render } from '@testing-library/react';

import Select from './select';

describe('Select', () => {
  const selectProps = {
    selectLabel: 'What is the reason for your visit?',
    placeholder: 'Select a motive',
    options: [
      {
        value: '61eea367ddf6c500149ae2cc',
        label: 'Cultural fit',
        meetingDuration: 30,
      },
      {
        value: '61eea350ddf6c500149ae2cb',
        label: 'Technical assessment',
        meetingDuration: 75,
      },
      {
        value: '61379ba159d4940022b6c929',
        label: 'Introduction Call',
        meetingDuration: 30,
      },
    ],
    onSelectionChange: jest.fn(),
    selectedOption: '61eea367ddf6c500149ae2cc',
  };
  it('should match snapshot', () => {
    const component = render(<Select {...selectProps} />);
    expect(component).toMatchSnapshot();
  });
});
