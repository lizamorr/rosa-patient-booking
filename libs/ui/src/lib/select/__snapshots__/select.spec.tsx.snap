// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="select"
      >
        <span
          class="select__label"
        >
          What is the reason for your visit?
        </span>
        <div
          class="select__dropdown-container"
        >
          <select
            class="select__dropdown"
            name="selectedOption"
            required=""
          >
            <option
              disabled=""
              hidden=""
              value=""
            >
              Select a motive
            </option>
            <option
              value="61eea367ddf6c500149ae2cc"
            >
              Cultural fit
            </option>
            <option
              value="61eea350ddf6c500149ae2cb"
            >
              Technical assessment
            </option>
            <option
              value="61379ba159d4940022b6c929"
            >
              Introduction Call
            </option>
          </select>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="select"
    >
      <span
        class="select__label"
      >
        What is the reason for your visit?
      </span>
      <div
        class="select__dropdown-container"
      >
        <select
          class="select__dropdown"
          name="selectedOption"
          required=""
        >
          <option
            disabled=""
            hidden=""
            value=""
          >
            Select a motive
          </option>
          <option
            value="61eea367ddf6c500149ae2cc"
          >
            Cultural fit
          </option>
          <option
            value="61eea350ddf6c500149ae2cb"
          >
            Technical assessment
          </option>
          <option
            value="61379ba159d4940022b6c929"
          >
            Introduction Call
          </option>
        </select>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
